src/OpenTelemetry.Instrumentation.Runtime/description.md

```diff
@@ -0,0 +1,198 @@
+Instrument Units [should follow](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-units)
+the Unified Code for Units of Measure.
+
+## GC related metrics
+
+The metrics in this section can be enabled by setting the
+`RuntimeMetricsOptions.IsGcEnabled` switch.
+
+| Name                                          | Description              | Units     | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|-----------------------------------------------|--------------------------|-----------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**gc.totalmemorysize** | Total allocated bytes    | `By`      | ObservableGauge   | `Int64`    |                  |                  |
+| process.runtime.dotnet.**gc.count**           | Garbage Collection count | `{times}` | ObservableCounter | `Int64`    | gen              | gen0, gen1, gen2 |
+
+Question for .NET team: is GC.GetTotalMemory(false) always equal to the sum of
+GC.GetGCMemoryInfo().GenerationInfo[i].SizeAfterBytes (i from 0 to 4)?
+
+I need to decide whether it makes sense to include both of them in the memory/GC
+size metrics.
+
+- [GC.GetTotalMemory](https://docs.microsoft.com/dotnet/api/system.gc.gettotalmemory):
+The number of bytes currently thought to be allocated.
+It does not wait for garbage collection to occur before returning.
+
+- [GC.CollectionCount](https://docs.microsoft.com/dotnet/api/system.gc.collectioncount):
+The number of times garbage collection has occurred for the specified generation
+of objects.
+
+### Additional GC metrics only available for NETCOREAPP3_1_OR_GREATER
+
+| Name                                              | Description                                      | Units | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|---------------------------------------------------|--------------------------------------------------|-------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**gc.allocated.bytes**     | Bytes allocated over the lifetime of the process | `By`  | ObservableCounter | `Int64`    |                  |                  |
+| process.runtime.dotnet.**gc.fragmentation.ratio** | GC fragmentation ratio                           | `1`   | ObservableGauge   | `Double`   |                  |                  |
+
+- [GC.GetTotalAllocatedBytes](https://docs.microsoft.com/dotnet/api/system.gc.gettotalallocatedbytes):
+  Gets a count of the bytes allocated over the lifetime of the process. The returned
+value does not include any native allocations. The value is an approximate count.
+
+- GC fragmentation ratio is calculated as:
+If `GC.GetGCMemoryInfo().HeapSizeBytes != 0`,
+the value is
+`GC.GetGCMemoryInfo().FragmentedBytes * 1.0d / GC.GetGCMemoryInfo().HeapSizeBytes`,
```

Comment:
Might need to call out that the result from `GetGCMemoryInfo` is used to get both `FragmentedBytes` and `HeapsizeBytes` so the behavior is transactional.

Reply:
I've reworded it to get the properties from calling `GetGCMemoryInfo` once.